Dockerfile

```diff
@@ -0,0 +1,40 @@
+#Another attempt at dockerizing frontend
+
+
+# Stage 1: Compile and Build angular codebase
+
+# Use official node image as the base image
+FROM node:14.17.3 as builder
+
+# Set the working directory
+WORKDIR /app
+
+# Add the source code to app
+COPY ./p3-frontend .
+
+# Install all the dependencies
+RUN npm ci
```

Comment:
it'd be better to copy the package*.json files over and npm ci first, so we could take advantage of build cache and skip that step in the future, but it's not practically important right now because (1) we're using ms-hosted agents that won't have any build cache, and (2) the devs are unlikely to use this file for local testing when `ng serve` is so easy to run.